### PR TITLE
Introduces a GATT client example

### PR DESCRIPTION
**Pull request by huntc**
This example compliments the [ble-gatt-server example PR](https://github.com/NordicSemiconductor/Android-BLE-Library/pull/234) by providing a client that is able to connect to it. Assuming that ble-gatt-server is running on another device, once this client connects it will display anything entered into the server. The example demonstrates best practices by running a foreground service that is able to automatically re-connect and display notifications to an encrypted characteristic. It also demonstrates how these notifications can manifest themselves through an Android binding.

Requires #234 to be merged. It should then be re-based.

TODO:

* [x] Not presently working due to the error code 133 problem [described by this issue](https://github.com/NordicSemiconductor/Android-BLE-Library/issues/53)

### CODE REVIEW

**Comment by philips77** [comment]
Once again thank you for that!
I'll do the code review today.

File: examples/ble-gatt-client/src/main/java/no/nordicsemi/android/ble/ble_gatt_client/GattService.kt [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        fun setMyCharacteristicChangedChannel(sendChannel: SendChannel<String>) {
+            myCharacteristicChangedChannel = sendChannel
+        }
+    }
+
+    private fun enableBleServices() {
+        val bluetoothManager = getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
+        if (bluetoothManager.adapter?.isEnabled == true) {
+            Log.i(TAG, "Enabling BLE services")
+            bluetoothManager.adapter.bondedDevices.forEach { device -> addDevice(device) }
```
Comment (huntc) [inline examples/ble-gatt-client/src/main/java/no/nordicsemi/android/ble/ble_gatt_client/GattService.kt:171]: If I bond a device via system settings, then should I expect that it should be connectable via BLE i.e. bonding information is the same no matter whether the pairing was done for BT classic or BLE on the proviso that the peripheral device supports BLE?

Comment (philips77) [reply examples/ble-gatt-client/src/main/java/no/nordicsemi/android/ble/ble_gatt_client/GattService.kt:171]: Yes. Bonding in Settings should I think be preferred way, add them the app doesn't need location permission and location service in order to scan for Ble devices.

File: examples/ble-gatt-client/src/main/java/no/nordicsemi/android/ble/ble_gatt_client/GattService.kt [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+            override fun initialize() {
+                setNotificationCallback(myCharacteristic).with { _, data ->
+                    if (data.value != null) {
+                        val value = String(data.value!!, Charsets.UTF_8)
+                        defaultScope.launch {
+                            myCharacteristicChangedChannel?.send(value)
+                        }
+                    }
+                }
```
Comment (huntc) [inline examples/ble-gatt-client/src/main/java/no/nordicsemi/android/ble/ble_gatt_client/GattService.kt:243]: Note that I had to move this callback outside of the queue statement below, where I previously used a `with` on `enableNotifications`. Could the fact that I had to do this be a bug?

**Comment by huntc** [comment]
@philips77 I've now tested the client and server code where the client runs on Android 10 and the server runs on Android 8.1. Everything appears to work. Could we please look to merging https://github.com/NordicSemiconductor/Android-BLE-Library/pull/234 ? I shall then convert this PR from a draft and rebase. Thanks.

**Comment by philips77** [comment]
#234 is merged. Go ahead and rebase that one. I'll merge it then. Once again thank you for your contribution!

**Comment by huntc** [comment]
@philips77 Now rebased and ready for a final review/merge.

**Comment by huntc** [comment]
> #234 is merged. Go ahead and rebase that one. I'll merge it then. Once again thank you for your contribution!

Hey there - will you be meeting then? Thanks.

**Comment by huntc** [comment]
Thanks @philips77 !

**Pull request merged**